src/controller/java/src/matter/controller/Messages.kt

```diff
@@ -0,0 +1,206 @@
+)
+
+/** Represents data received from a read operation. */
+sealed class ReadData {
+  /**
+   * Represents data related to an event.
+   *
+   * @param path The event path associated with the data.
+   * @param eventNumber the event number value that is scoped to the node
+   * @param priorityLevel the priority describes the usage semantics of the event
+   * @param timeStamp the timestamp at the time the event was created
+   * @param data The ByteArray containing the data in TLV format.
+   */
+  class Event(
+    val path: EventPath,
+    val eventNumber: ULong,
+    val priorityLevel: UByte,
+    val timeStamp: Timestamp,
+    val data: ByteArray
+  ) : ReadData()
+
+  /**
+   * Represents data related to an attribute.
+   *
+   * @param path The attribute path associated with the data.
+   * @param data The ByteArray containing the data in TLV format.
+   */
+  class Attribute(val path: AttributePath, val data: ByteArray) : ReadData()
+}
+
+/** Represents a failure that can occur during a read operation. */
+sealed class ReadFailure {
+  /**
+   * Represents a failure related to an event path.
+   *
+   * @param path The event path information associated with the failure.
+   * @param error The exception that describes the failure.
+   */
+  class Event(val path: EventPath, val error: Exception) : ReadFailure()
+
+  /**
+   * Represents a failure related to an attribute path.
+   *
+   * @param path The attribute path information associated with the failure.
+   * @param error The exception that describes the failure.
+   */
+  class Attribute(val path: AttributePath, val error: Exception) : ReadFailure()
+}
+
+/**
+ * Represents the response from a read operation, containing both successes and failures.
+ *
+ * @param successes A list of successfully read data elements.
+ * @param failures A list of failures that occurred during the read operation.
+ */
+class ReadResponse(val successes: List<ReadData>, val failures: List<ReadFailure>)
+
+/**
+ * Information about a subscribe request element.
+ *
+ * @param eventPaths A list of event path information in the read request.
+ * @param attributePaths A list of attribute path information in the read request.
+ * @param minInterval The minimum interval boundary floor in seconds.
+ * @param maxInterval The maximum interval boundary ceiling in seconds.
+ * @param keepSubscriptions Indicates whether to keep existing subscriptions.
+ * @param fabricFiltered Limits the data read within fabric-scoped lists to the accessing fabric.
+ */
+class SubscribeRequest(
+  val eventPaths: List<EventPath>,
+  val attributePaths: List<AttributePath>,
+  val minInterval: Duration = Duration.ofSeconds(SUBSCRIPTION_MIN_INTERVAL_S),
+  val maxInterval: Duration = Duration.ofSeconds(SUBSCRIPTION_MAX_INTERVAL_S),
+  val keepSubscriptions: Boolean = true,
+  val fabricFiltered: Boolean = true
+)
+
+/** An interface representing the possible states of a subscription. */
+sealed interface SubscriptionState {
```

Comment:
all the others are `sealed class` but this is an interface. Any reason to pick one over the other?

if there is a reason, maybe add a comment of why this seems the odd one out. If there is no reason, being consistent throughout seems to be better.

Reply:
There is no specific reason, we need to initialize the constructor of the parent sealed class if change to sealed class.
class SubscriptionErrorNotification(val terminationCause: UInt) : SubscriptionState
to
class SubscriptionErrorNotification(val terminationCause: UInt) : SubscriptionState()

Change to sealed class to keep consistent